src/CoreBundle/private/library/classes/TCMSResourceCollection.class.php

```diff
@@ -69,98 +81,161 @@ public function IsEnabled()
      */
     public function IsAllowed()
```

Comment:
Is this (already) checked somewhere else?

What does happen when there are two concurrent requests with the second request?

Reply:
IsAllowed is called by the core - to decided if to trigger resource collection. It had no concept of concurrent calls so it would return true for every request wanting to generate resources. That is the change I made - that it now only allows resource collection if no other process on that node is currently writing resources